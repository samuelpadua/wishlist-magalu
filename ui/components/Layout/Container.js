import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Box } from 'rebass'

const ContainerStyled = styled(Box)`
  position: relative;
  margin: 0 auto !important;
  max-width: 1220px;
`

function Container({ children, ...props }) {
  return (
    <ContainerStyled
      {...props}
      px={15}
      width={props.width}
    >
      {children}
    </ContainerStyled>
  )
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  width: PropTypes.string
}

Container.defaultProps = {
  width: '100%'
}

export default Container
