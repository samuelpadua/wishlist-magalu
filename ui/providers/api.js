import Axios from 'axios'

export const urls = {
  test: `http://localhost`,
  development: 'http://localhost:9000',
  production: 'http://localhost:9000'
}

const api = Axios.create({
  baseURL: urls[process.env.NODE_ENV],
  headers: {
    'Accept': 'application/json',
    'Content-Type': 'application/json'
  }
})

export default api
