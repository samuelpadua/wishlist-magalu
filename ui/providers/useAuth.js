import React, { createContext, useState, useContext, useEffect } from 'react'
import Cookies from 'js-cookie'
import Router from 'next/router'

import api from './api'

export const AuthContext = createContext({});

export function AuthProvider({ children }) {
  const [authenticated, setAuthenticated] = useState(null)
  const [loading, setLoading] = useState(true)
  const [customer, setCustomer] = useState(null)

  useEffect(() => {
    async function validateLoggedCustomer() {
      const token = Cookies.get('token')
      if (token) {
        api.defaults.headers.authorization = token

        const { data } = await api.get('/api/auth/me')

        if (data) {
          setCustomer(data)
          setAuthenticated(true)
        }
      }

      setLoading(false)
    }

    validateLoggedCustomer()
  }, [])

  async function login(email) {
    const response = await api.post('/api/auth', { email })
    const { data: { token } } = response

    if (token) {
      Cookies.set('token', token, { expires: 60 })
      api.defaults.headers.authorization = token

      const { data } = await api.get('/api/auth/me')

      setCustomer(data)
      setAuthenticated(true)
      setLoading(false)
    }
  }

  function logout() {
    Cookies.remove('token')
    setAuthenticated(false)
    setCustomer({})

    Router.push('/')
  }

  return (
    <AuthContext.Provider value={{ isAuthenticated: authenticated, login, customer, loading, logout }}>
      {children}
    </AuthContext.Provider>
  )
}

export function ProtectRoute(Component) {
  return () => {
    const { isAuthenticated, loading } = useAuth()

    useEffect(() => {
      if (!isAuthenticated && !loading) {
        Router.push('/entrar')
      }
    }, [loading, isAuthenticated])

    return (<Component {...arguments} />)
  }
}

export default function useAuth() {
  const context = useContext(AuthContext)

  return context
}

