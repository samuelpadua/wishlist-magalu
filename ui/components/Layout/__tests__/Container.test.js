import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { AuthProvider } from '../../../providers/useAuth'
import Container from '../Container'

describe('Container component', () => {
  it('should render no logged customer', () => {
    const tree = renderer
      .create(<Container>Content</Container>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
