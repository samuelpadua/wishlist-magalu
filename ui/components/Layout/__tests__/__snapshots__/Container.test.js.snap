// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render no logged customer 1`] = `
.c0 {
  position: relative;
  margin: 0 auto !important;
  max-width: 1220px;
}

<div
  className="c0 css-5g036t"
>
  Content
</div>
`;
